### PR TITLE
CN-2581 Create CoinPicker screen

### PR DESCRIPTION
**Pull request by nemanjakrstonic**
PR Checklist:
- [x] I have named the PR correctly
- [x] I have added at least two reviewers
- [x] I have assigned the PR to myself
- [x] I have merged develop into my feature branch and resolved conflicts
- [x] I have tested all new features locally
- [x] I have tested relevant flow locally

### How to name a PR
- If it includes only one issue [CN-1234 Issue name]
- If it includes many issues [CN-1234, CN-2345, CN-3456 What was updated/changed/added]
- If the PR is not ready yet, add [DRAFT] in front of the name

### How to test locally
1. Test happy path
2. Test other paths
3. Remove all warnings

### How to relevant flow locally
- If working on part of CelPay, check the whole happy path

### CODE REVIEW

File: app/components/molecules/CoinPicker/CoinPicker.js [hunk 1]
```diff
@@ -25,39 +26,84 @@ class CoinPicker extends Component {
...
       onCoinSelect(field, defaultSelected);
       updateFormField(field, defaultSelected);
     }
   }
 
+  renderByType = () => {
+    const { type, value, onCoinSelect, navigateTo, field, coinCompliance } = this.props;
+    const { coinListFormated } = this.state
+
+    const coinFullName = coinCompliance
```
Comment (StefanBurscher) [inline app/components/molecules/CoinPicker/CoinPicker.js]: Calculate this only if type is enterAmount

File: app/components/molecules/CoinPicker/CoinPicker.js [hunk 2]
```diff
@@ -25,39 +26,82 @@ class CoinPicker extends Component {
...
+        )
+
+
+      case "enterAmount":
+        return (
+          <>
+            <TouchableOpacity onPress={() => navigateTo('SelectCoin', { coinList: coinListFormated, onCoinSelect, field })}>
+              <View style={{ alignContent: 'center', alignItems: 'center', alignSelf: 'center', flexDirection: 'row', paddingHorizontal: 5, paddingVertical: 5 }}>
+                <CelText type="H3" style={{ paddingRight: 10, }}>
+                  {coinCompliance.filter(coin => coin.value === value).map(coin => coin.label)}
```
Comment (StefanBurscher) [inline app/components/molecules/CoinPicker/CoinPicker.js]: this is bad... You are interacting through the array two time without any reason...

File: app/components/organisms/CelHeading/CelHeading.js [hunk 3]
```diff
@@ -159,14 +159,13 @@ class CelHeading extends Component {
...
-
     return (
       <View style={[style.content, paddings]}>
         <View style={leftStyle}>
           {this.getLeftContent(scene.options)}
           {formData.activeSearch && (
-            <View style={[{ width: '100%', justifyContent: 'center', alignSelf: 'center', marginLeft: 12 }]}>
-              <CelInput debounce autoFocus={formData.activeSearch} basic margin="0 0 0 0" field="search" placeholder="Dialing code, country…" type='text' value={this.props.formData.search} />
+            <View style={[{ width: '100%', justifyContent: 'center', paddingTop: 20, alignSelf: 'center', marginLeft: 12 }]}>
+              <CelInput debounce autoFocus={formData.activeSearch} basic margin="0 0 0 0" field="search" placeholder={scene.state.routeName === 'SelectCoin' ? "Choose coin" : "Dialing code, country…"} type='text' value={this.props.formData.search} />
```
Comment (StefanBurscher) [inline app/components/organisms/CelHeading/CelHeading.js]: SelectCoin should be probably Select coin

File: app/components/screens/SelectCoin/SelectCoin.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+  static navigationOptions = () => (
+    {
+      title: "Select Coin",
+      right: 'search'
+    }
+  )
+
+  static getDerivedStateFromProps(nextProps) {
+
```
Comment (StefanBurscher) [inline app/components/screens/SelectCoin/SelectCoin.js]: Potential memory leak and app general slowness. Please add an if for checking if setState is necessary.
Something like (!prevState.filteredCoins || prevState.filteredCoins.length === 0)

File: app/components/screens/SelectCoin/SelectCoin.js [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    // const newState = {};
+    const coinList = nextProps.navigation.getParam('coinList')
+    const allCoins = nextProps.currencies.filter(item => coinList.includes(item.short));
+
+    // const field = nextProps.navigation && nextProps.navigation.getParam('coinList');
```
Comment (StefanBurscher) [inline app/components/screens/SelectCoin/SelectCoin.js]: Delete unnecessary comments

File: app/components/screens/WithdrawEnterAmount/WithdrawEnterAmount.js [hunk 6]
```diff
@@ -296,13 +295,14 @@ class WithdrawEnterAmount extends Component {
...
-                  items={coinSelectItems}
-                  field='coin'
-                  displayValue={coin}
+
+                <CoinPicker
+                  type={'enterAmount'}
                   updateFormField={actions.updateFormField}
-                  onChange={this.handleCoinChange}
-                  placeholder='Choose a coin'
+                  value={formData.coin}
```
Comment (StefanBurscher) [inline app/components/screens/WithdrawEnterAmount/WithdrawEnterAmount.js]: use coin instead of formData.coin, like it was before

Comment (StefanBurscher) [reply app/components/screens/WithdrawEnterAmount/WithdrawEnterAmount.js]: Use this.handleCoinChange!

File: app/components/molecules/CoinPicker/CoinPicker.js [hunk 7]
```diff
@@ -1,12 +1,9 @@
...
-
-
-
-import SimpleSelect from '../SimpleSelect/SimpleSelect';
+import { View, TouchableOpacity } from 'react-native';
 import CelText from '../../atoms/CelText/CelText';
 import CircleButton from '../../atoms/CircleButton/CircleButton';
+import Icon from '../../atoms/Icon/Icon';
 
 class CoinPicker extends Component {
```
Comment (StefanBurscher) [inline app/components/molecules/CoinPicker/CoinPicker.js]: Implement it on BorrowEnterAmount as well

File: app/components/organisms/CelHeading/CelHeading.js (hunk 3 above)
Comment (nemanjakrstonic) [reply app/components/organisms/CelHeading/CelHeading.js]: Actually no, scene.state.routeName equals to `SelectedCoin`

File: app/components/screens/BorrowEnterAmount/BorrowEnterAmount.js [hunk 8]
```diff
@@ -189,16 +191,18 @@ class BorrowEnterAmount extends Component {
...
-                value={formData.coin}
-                updateFormField={actions.updateFormField}
-                onChange={(value) => actions.updateFormFields({ coin: value, loanType: value === 'USD' ? LOAN_TYPES.USD_LOAN : LOAN_TYPES.STABLE_COIN_LOAN})}
-                placeholder='Choose a coin'
-              />
+            {isCompanyMember() && (
+              <View style={styles.selectWrapper}>
+                <CoinPicker
+                  type={'enterAmount'}
+                  onChange={(value) => actions.updateFormFields({ coin: value, loanType: value === 'USD' ? LOAN_TYPES.USD_LOAN : LOAN_TYPES.STABLE_COIN_LOAN })}
```
Comment (StefanBurscher) [inline app/components/screens/BorrowEnterAmount/BorrowEnterAmount.js]: You have not implemented the onChange prop handling in the CoinPicker component

File: app/components/screens/SelectCoin/SelectCoin.js [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+  static getDerivedStateFromProps(nextProps, prevState) {
+
+    const coinList = nextProps.navigation.getParam('coinList')
+    const allCoins = nextProps.currencies.filter(item => coinList.includes(item.short));
+
+    const text = nextProps.formData && nextProps.formData.search && nextProps.formData.search.toLowerCase() || "";
+
+    const filteredCoins = allCoins.filter(coin => (coin.name.toLowerCase().includes(text) || coin.short.toLowerCase().includes(text)))
+
+    if (!prevState.filteredCoins || prevState.filteredCoins.length === 0) {
```
Comment (StefanBurscher) [inline app/components/screens/SelectCoin/SelectCoin.js]: The issue is not solved, you still calculate everything on every change but now, you just don't set it in state. You should move this condition on the top of the getDerivedStateFromProps function, on the place where I put you the comment last time as well...

**Pull request merged**